### DISCUSSION TITLE
How to prevent resizing the cmdline (cmdheight)?

### DISCUSSION THREAD

**Discussion by bvngee** [mrjones2014/smart-splits.nvim#167] [Q&A]
Hey all, I think this is more of a general neovim question but I'm deciding to ask here in case anyone has experience with this using this plugin. 
The problem I'm having is that any resize command (eg. resize_up when in a single empty buffer) also changes the value of `cmdheight`, thus expanding the cmdline window size when I dont want it to (I want it to stay at cmdheight=0). Does anyone know if it's possible to prevent this (eg. is there a filetype/buftype I could add that would pull this off)?

**Comment by mrjones2014** [comment]
That doesn't happen for me 🤔

**Reply by bvngee** [reply]
Hmm... I can upload a video of it happening as well as my config if either would help. 
Video: https://streamable.com/ersfz2 
Config (using lazy.nvim):
```lua
return {
  'mrjones2014/smart-splits.nvim',
  dependencies = {
    'kwkarlwang/bufresize.nvim',
  },
  event = 'VeryLazy',
  config = function()
    local smart_splits = require('smart-splits')
    local opts = {
      ignored_buftypes = {
        'nofile',
        'quickfix',
        'qf',
        'prompt',
      },
      ignored_filetypes = { 'NvimTree' },
      default_amount = 2,
      at_edge = 'stop',
      move_cursor_same_row = false,
      cursor_follows_swapped_bufs = true,
      resize_mode = {
        quit_key = '<ESC>',
        resize_keys = { 'h', 'j', 'k', 'l' },
        silent = true,
        hooks = {
          on_enter = function()
            vim.notify('Entering resize mode: use h/j/k/l to resize')
            vim.keymap.set('n', 'H', '5h', { remap = true })
            vim.keymap.set('n', 'L', '5l', { remap = true })
            vim.keymap.set('n', 'J', '5j', { remap = true })
            vim.keymap.set('n', 'K', '5k', { remap = true })
          end,
          on_leave = function()
            vim.notify('Exiting resize mode, all keymaps restored')
            vim.keymap.del('n', '<S-h>')
            vim.keymap.del('n', '<S-l>')
            vim.keymap.del('n', '<S-j>')
            vim.keymap.del('n', '<S-k>')

            require('bufresize').register()
          end,
        },
      },
      ignored_events = {
        'BufEnter',
        'WinEnter',
      },
      multiplexer_integration = nil,
      log_level = 'info',
    }

    smart_splits.setup(opts)

    -- swapping buffers between windows
    vim.keymap.set('n', '<C-A-h>', smart_splits.swap_buf_left)
    vim.keymap.set('n', '<C-A-l>', smart_splits.swap_buf_right)
    vim.keymap.set('n', '<C-A-j>', smart_splits.swap_buf_down)
    vim.keymap.set('n', '<C-A-k>', smart_splits.swap_buf_up)

    -- persistent rezize mode
    vim.keymap.set('n', '<leader>rs', smart_splits.start_resize_mode)
  end,
}
```
the rest of my nvim config (including options and stuff) is [here](https://github.com/BvngeeCord/nixconf/tree/main/home/nvim_new/lua/user) if that helps.

Lmk if there's anything that sticks out to you as obvious or if you have ideas for what I should look into. Thanks for the awesome plugin!!

**Reply by bvngee** [reply]
As I have switched to using noice.nvim which overrides the `cmdline`, I no longer have this issue myself - but it still occurs with the above configuration, and as I'm now realizing, with a blank slate neovim as well. `nvim -u /dev/null` followed by `<C-w>-` increases `cmdheight`.
